src/metric/disk.rs

```diff
@@ -1,13 +1,16 @@
-use crate::{util, Agent, Values};
+use super::{HostMetric, HostMetricKind, MetricValue};
+use crate::{util, Agent};
 use os_stat::Disk;
 use std::{collections::HashMap, time::Duration};
 
-impl Agent {
-    pub fn get_disk_metrics() -> Option<Values> {
-        let interval = Duration::from_secs(10);
-        let previous = Disk::get().expect("failed to get disk statistics");
-        std::thread::sleep(interval);
-        let current = Disk::get().expect("failed to get disk statistics");
+type Disks = Vec<Disk>;
+
+// TODO: investigate how many seconds are used in mackerel-agent
+const INTERVAL: Duration = Duration::from_secs(10);
+
+impl From<(Disks, Disks)> for HostMetric {
+    fn from((previous, current): (Disks, Disks)) -> Self {
```

Comment:
This is testable, so it's better to write tests. But it's not necessary in this pull request.